cmd/coder/main.go

```diff
@@ -15,6 +20,48 @@ import (
 func main() {
 	rand.Seed(time.Now().UnixMicro())
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Add a custom SIGQUIT handler that outputs to stderr and a well-known file
+	// in the home directory. This also prevents SIGQUITs from killing the CLI.
+	go func() {
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGQUIT)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-sigs:
+			}
+
+			buf := make([]byte, 10_000_000)
+			stacklen := runtime.Stack(buf, true)
+
+			_, _ = fmt.Fprintf(os.Stderr, "SIGQUIT:\n%s\n", buf[:stacklen])
+
+			// Write to a well-known file.
+			dir, err := os.UserHomeDir()
+			if err != nil {
+				dir = os.TempDir()
+			}
+			fpath := filepath.Join(dir, fmt.Sprintf("coder-agent-%s.dump", time.Now().Format("2006-01-02T15:04:05.000Z")))
```

Comment:
I think these files are usually written to the current working directory, is there a reason we don't want to do that here?

Reply:
I can understand the use-case for workspaces, I think we have more flexibility in dumping into home there. Should we at least move it to the agent cmd in that case?

The loss register values and supporting `GOTRACEBACK` is unfortunate, though. We could at least enable `GOTRACEBACK` by conditionally starting this routine only if it's unset.

Reply:
I disabled listening for SIGQUIT if `GOTRACEBACK=crash`, is that good?